solutions/platinum/leetcode-149.mdx

```diff
@@ -0,0 +1,54 @@
+---
+id: leetcode-149
+source: Leetcode
+title: Max Points on a Line
+author: Mihnea Brebenel
+---
+
+## Explanation
+
+Because $n \le 300$ we can use a brute-force approach: choose two points and check if other point are collinear with them.
+Three points $A$, $B$, and $c$ are collinear if and only if $AB$ has the same slope as $BC$:
+
+$$
+\frac{A.y-B.y}{A.x-B.x}=\frac{B.y-C.y}{B.x-C.x}
+$$
+
+In practice you'll find this formula in a product form to avoid dealing with floating-point values.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(n^3)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+class Solution {
+  public:
+	int maxPoints(vector<vector<int>> &points) {
+		if (points.size() <= 2) { return (int)points.size(); }
+
+		int ans = 0;
+		for (int i = 0; i < points.size(); i++) {
+			for (int j = i + 1; j < points.size(); j++) {
+				int p = 0;
+				for (int k = j + 1; k < points.size(); k++) {
+					if ((points[j][1] - points[i][1]) *
+					        (points[i][0] - points[k][0]) ==
+					    (points[i][1] - points[k][1]) *
+					        (points[j][0] - points[i][0])) {
```

Comment:
put the expressions in this if statement into vars like `lhs` and `rhs` and add a comment "check if [smth] per the formula given above"

Reply:
done